webapp/src/containers/wordpress/index.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import fetchData from '../../lib/fetchData';
+import { fetchWp, fetchWpSuccess, fetchWpFailure } from '../../actions/wp';
+import { selectWp } from '../../selectors/wpSelectors';
+
+import HeadMetaTags from '../../components/headMetaTags';
+import{ WP_REST_API_BASE_URL, MENU } from '../../constants';
+
+
+class Wordpress extends Component {
+  constructor(props){
+    super(props);
+    this.currentRoute='';
+  }
+  componentDidMount() {
+    let currentRoute=this.getCurrentRoute(this.props);
+    this.currentRoute=currentRoute;
+    this.getData(currentRoute);
+  }
+  componentWillUpdate(nextProps, nextState){
+    let currentRoute=this.getCurrentRoute(nextProps);
+    if(this.currentRoute != currentRoute){
+      this.currentRoute=currentRoute;
```

Comment:
As discussed, let's see if `this.getCurrentRoute(this.props) !== this.getCurrentRoute(nextProps) ` works, in which case we would be able to avoid this.currentRoute and this.currentState. Thanks!